### ISSUE TITLE
Shopify auth servers incorrectly reporting 403.

### ISSUE THREAD

**Issue by YourWishes** [Shopify/shopify-cli#1694]
# Issue summary

Attempting to run theme commands is rejected due to authorization errors.

## Expected behavior

Executing `shopify theme pull` should simply pull the theme.

## Actual behavior

- Following the stack the issue is during the API requests to the Shopify servers.
- This seems to be somewhat intermittent and seems to be Shopify store dependent from time-to-time.
- Our team has only been experiencing the issue since the 25th of October
- Switching to a different store is working, and I can execute theme commands on most other stores
- Different team members are having issues for different stores, e.g. I could authenticate with store-a but my colleague couldn't, however, I cannot authenticate with store-b yet my colleague can. And we can both authenticate with store-c.
- Full cache clearing, cookie clearing, logout and re-login does not solve the problem.
- I can confirm that the access token is being regenerated within the CLI each time I login/logout, yet still am not authorized

See error screenshots;
![image](https://user-images.githubusercontent.com/14732069/139131347-50058d15-44e1-465d-9076-9da3dae9523a.png)

![image](https://user-images.githubusercontent.com/14732069/139131541-48bf06a6-4d42-4a72-9729-17625f1d16c2.png)

## Steps to reproduce the problem

Exact steps are a bit uncertain since it's so sporadic

1. Logout from CLI if logged in
2. Log in using a staff admin account to a development store
3. Execute any theme command, typically `shopify theme pull` or `shopify theme push`
4. Chance of receiving an authentication issue despite having authenticated

## Specifications

- App type: Shopify CLI, Theme commands
- Operating System: Windows 10 Home 10.1.1903, other team members are running some form of Mac OS
- Shell: Powershell, Term on Mac
- Ruby version (ruby -v): ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x64-mingw32]
- Shopify CLI version: 2.6.5, also have tried a few older versions to no avail

## Additional Information
- I can confirm that the login I am using is the staff account, not the partner account. If it's a potential issue they are the same email but I am using the Single-Sign-On account
- `shopify whoami` is correctly reporting my access prior to running theme commands.
- Tonnes of testing has lead me to believe that _partner_ accounts seem to always work on production/non-development stores, and staff accounts seem to always work on development stores.

**Comment by pepicrft** [comment]
Hey @YourWishes & @meewgumi 
I'm very sorry you are running into this inconvenience with the CLI's authentication. The team will soon work on solving this on the server-side so that you don't have to take those manual steps. Until we have that fix rolled out, I recommend following @YourWishes's steps as a valid workaround.

**Comment by pmourer-zaneray** [comment]
tl;dr

Create a staff account that is your actual email that is registered in the partner's dashboard if your organization has a different email. The store owner on a newly created dev store will have the organizations email, not your user's. 

I hadn't had this problem at all and it was because I was freelancing so the organization email was also my partners login email. If you create a dev store from the dashboard, the store owner will have the organization's email and your login email is not at all associated with the account so this issue sort of makes sense from that perspective.

This really needs to be called out somewhere obvious.

**Comment by tpage99** [comment]
Having this same issue but only on one dev store. Still have access to all other dev stores and no issues with `push` or `pull` it's just the one store. 

Getting the `block in request': 403 (ShopifyCLI::API::APIRequestForbiddenError)
{"errors":"Unauthorized Access"}`

Haven't tried the creating a staff account workaround, which seems pretty silly given I'm using the same pattern in other dev stores and not having an issue at all. 

Shopify version 2.15.2 and ruby 3.0.0
